system/security/securesocket/securesocket.hpp

```diff
@@ -30,6 +30,7 @@
 
 #include "jsocket.hpp"
 #include "jptree.hpp"
+#include "jsmartsock.ipp"
```

Comment:
securesocket.hpp doesn't need the jsmartsock.ipp but does need ISmartSocketFactory, so:
1) include jsmartsock.hpp in securesocket.hpp (for ISmartSocketFactory)
2) include jsmartsock.ipp in securesocket.cpp (for CSmartSocketFactory / CSmartSocket)